dom/src/main/scala/fs2/dom/MediaSource.scala

```diff
@@ -0,0 +1,303 @@
+package fs2.dom
+
+import org.scalajs.dom
+import scala.scalajs.js
+import cats.effect._
+import cats.syntax.all._
+
+import scala.scalajs.js.annotation.*
+
+object Url {
+
+  private def createObjectUrl[F[_]](
+      objectUrl: F[String]
+  )(implicit F: Async[F]): Resource[F, Url] =
+    Resource
+      .make(objectUrl)(url => F.delay(dom.URL.revokeObjectURL(url)))
+      .map(url => new dom.URL(url))
+
+  def createObjectUrl[F[_]](blob: dom.Blob)(implicit F: Async[F]): Resource[F, Url] =
+    createObjectUrl(F.delay(dom.URL.createObjectURL(blob)))
+
+  def createObjectUrl[F[_]](src: MediaSource[F])(implicit F: Async[F]): Resource[F, Url] =
+    createObjectUrl(F.delay(dom.URL.createObjectURL(src)))
+}
+
+import scala.scalajs.js.typedarray.ArrayBuffer
+
+@JSGlobal
+@js.native
+class SourceBuffer[F[_]] protected () extends js.Any
+object SourceBuffer {
```

Comment:
I haven't studied these APIs yet, but generally I prefer using wrappers instead of the facade trick. The main motivation for using the facade trick for `HtmlElement` and friends is to reduce allocations / memory pressure since I expect that an app may make a lot of those, sometimes very quickly.

Reply:
Sure, the source buffers are created too much; I was intending on using the facade trick so we could still cast from `SourceBuffer` to `dom.SourceBuffer` if the API wasn't complete. I guess we could expose the underlying field if needs be?

Reply:
> I guess we could expose the underlying field if needs be?

Ah, yes you can do some tricks like that. For example:

https://github.com/yilinwei/fs2-dom/blob/af450dc9dafd4935a73692a2971d5086ee334699/dom/src/main/scala/fs2/dom/Event.scala#L55

if you have an `event: Event[IO]` you can access the underlying event like `event.asInstanceOf[{ def event: dom.Event }].event`